File: PYTHON/ArrayAndStrings/Easy/MoveZeros.py
def moveZeros(arr):
    non_zero_index = 0
    for i in range(len(arr)):
        if arr[i] != 0:
            arr[non_zero_index] = arr[i]
            non_zero_index += 1
    for i in range(non_zero_index, len(arr)):
        arr[i] = 0
    return arr

# Example usage:
if __name__ == "__main__":
    arr = [0, 1, 0, 3, 12]
    print("Original array:", arr)
    result = moveZeros(arr)
    print("Array after moving zeros:", result)

File: JAVASCRIPT/ArrayAndStrings/Easy/MoveZeros.js
// MOVE ZEROS

function moveZero(arr){
    let nonZero = 0
    for(let i=0;i<arr.length;i++){
        if(arr[i] !== 0){
            arr[nonZero] = arr[i]
            nonZero++;
        }
    }
    for(let i = nonZero; i < arr.length; i++){
        arr[i] = 0;
    }
    return arr;
}


// Example usage:
const arr = [0, 1, 0, 3, 12];
const result = moveZero(arr);
console.log(result); // Output: [1, 3, 12, 0, 0]